WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -20,7 +20,3 @@
     </div>
   </div>
 </div>
-
-<script>
-  $('#competitions-<%= params[:display] == "list" ? "map" : "list" %>').hide();
-</script>
```

Comment:
The index.js.erb is loaded only after Ajax request, but the first time you load the page html request is sent so the script wouldn't be executed.